Update Table and AdvancedTable content to make documentation more concise (#3187)
hashicorp/design-system: website/docs/components/table/advanced-table/partials/guidelines/guidelines.md

> ### Sorting

Reviewer: Remove "sorting" from the warning under the multi-select section and add a warning to the Sorting section.
 
```suggestion
### Sorting

!!! Warning

**Unsupported feature**

Sorting is not supported for nested rows at this time.
!!!
```

The row placement property is only relevant within Figma and doesn’t exist as a property within the code.
!!!

Row placement determines the visual styling based on where the row is placed. Only cells with a column placement that is either `start` or `end` use the row placement property; column position `middle` does not use this property.

![The cell with column placement end and row placement end has a border radius set on the bottom right corner.](/assets/components/table/advanced-table/table-row-placement.png)

> ## Cells

Reviewer: [suggestion] There's additional content under the Cells header in the Table docs that feels particularly helpful. Should we consider adding it to the Advanced Table docs as well?

> For the user to scan, sort, and filter the table easily, each cell should contain a single piece of data. Having more than one piece of data in a cell makes it harder for users to navigate the relationships between headers and cells.

![An Advanced Table with a pinned column with the context menu open and a single "Unpin column" option](/assets/components/table/advanced-table/advanced-table-unpin-column.png)

## Multi-Select

!!! Warning

**Unsupported feature**

> Multi-select and sorting are not supported for nested rows at this time.

Reviewer: ```suggestion
Multi-select is not supported for nested rows at this time.
```
